src/Shared/PackageManagers/BaseProjectManager.cs

```diff
@@ -349,6 +351,32 @@ public async Task<string> ExtractArchive(string directoryName, byte[] bytes, boo
             return fullExtractionPath;
         }
 
+        /// <summary>
+        ///     Gets the latest version from the package metadata
+        /// </summary>
+        /// <param name="metadata"> </param>
+        /// <returns> </returns>
+        public Version? GetLatestVersion(JsonDocument metadata)
+        {
+            List<Version> versions = GetVersions(metadata);
+            return GetLatestVersion(versions);
+        }
+
+        /// <summary>
+        ///     overload for getting the latest version
+        /// </summary>
+        /// <param name="versions"> </param>
+        /// <returns> </returns>
+        public Version? GetLatestVersion(List<Version> versions)
+        {
+            if (versions != null && versions?.Count > 0)
```

Comment:
```suggestion
            if (versions?.Count > 0)
```
You don't need the first null check with the ? I believe.